spring-cloud-gcp-data-spanner/src/main/java/com/google/cloud/spring/data/spanner/core/convert/StructAccessor.java

```diff
@@ -104,13 +104,19 @@
 
   private Set<String> columnNamesIndex;
 
-  private static final Gson gson = new Gson();
+  private Gson gson;
 
   public StructAccessor(Struct struct) {
```

Comment:
I was looking for where the single-arg version was still used, and now I wonder if there is an issue with `ConverterAwareMappingSpannerEntityReader` not being able to process JSON fields. Could you figure out what path `ConverterAwareMappingSpannerEntityReader.read()` goes through?

My original thought was to deprecate the single-arg version and stop using it, since having Gson object be null in some cases but not others could easily lead to bugs.

Reply:
hmm, the single-arg version is now only used in `SqlSpannerQuery` [here](https://github.com/GoogleCloudPlatform/spring-cloud-gcp/blob/98756c2ffd27ad207aca405290d90acf33a2752c/spring-cloud-gcp-data-spanner/src/main/java/com/google/cloud/spring/data/spanner/repository/query/SqlSpannerQuery.java#L241). it will make no difference to functionality to pass a second gson parameter in. So I think deprecating makes sense.

As for `ConverterAwareMappingSpannerEntityReader.read()` , it eventually processes each property of entity in `StructPropertyValueProvider` and JSON fields are read with [this line](https://github.com/GoogleCloudPlatform/spring-cloud-gcp/blob/main/spring-cloud-gcp-data-spanner/src/main/java/com/google/cloud/spring/data/spanner/core/convert/StructPropertyValueProvider.java#L115). `this.structAccessor` here is passed down from `ConverterAwareMappingSpannerEntityReader` which has access to `SpannerMappingContext` and thus `Gson` bean.